src/serializer/ResidualFunctions.js

```diff
@@ -174,22 +183,28 @@ export class ResidualFunctions {
     }
   }
 
-  _getReferentializedScopeInitialization(scope: ScopeBinding): BabelNodeIfStatement {
+  _getReferentializedScopeInitialization(scope: ScopeBinding): [BabelNodeVariableDeclaration, BabelNodeIfStatement] {
     let properties = [];
     for (let [variableName, value] of scope.initializationValues.entries()) {
       properties.push(t.objectProperty(t.identifier(variableName), value));
     }
-
-    return t.ifStatement(
-      t.unaryExpression("!", t.memberExpression(this.capturedScopesArray, t.identifier(scope.name), true)),
-      t.expressionStatement(
-        t.assignmentExpression(
-          "=",
-          t.memberExpression(this.capturedScopesArray, t.identifier(scope.name), true),
-          t.objectExpression(properties)
+    let initExpression = t.memberExpression(this.capturedScopesArray, t.identifier(scope.name), true);
+    let capturedScope = this.capturedScopeGenerator.deriveFrom(scope.name);
```

Comment:
Instead of calling deriveFrom, just get another name via the already existing scopeNameGenerator.